tea.go

```diff
@@ -281,66 +293,83 @@ func Interrupt() Msg {
 }
 
 // NewProgram creates a new Program.
-func NewProgram(model Model, opts ...ProgramOption) *Program {
-	p := &Program{
-		initialModel: model,
-		msgs:         make(chan Msg),
-		rendererDone: make(chan struct{}),
-		keyboardc:    make(chan struct{}),
-		modes:        ansi.Modes{},
-	}
+func NewProgram[T any](model Model[T]) *Program[T] {
+	p := new(Program[T])
+	p.Init = model.Init
+	p.Update = func(t T, msg Msg) (T, Cmd) { return any(t).(Model[T]).Update(msg) }
```

Comment:
This line would me much simpler if `Model[T]` was a struct instead of an interface. See https://github.com/charmbracelet/bubbletea/issues/1136#issuecomment-2754104666